src/parser.y

```diff
@@ -215,13 +216,13 @@ assignment:
                         auto prevFile = assignment.location().fileName();
                         auto currFile = LOC(@$).fileName();
                         
-                        if(assignmentWarning && prevFile==RootFile && currFile == RootFile){
+                        if(bMainModule && assignmentWarning && prevFile==RootFile && currFile == RootFile){
```

Comment:
`assignmentWarning && prevFile==RootFile` is shared between the if and else branches. Pulling this out could make the code a bit more readable.